plansys2_optic_plan_solver/src/plansys2_optic_plan_solver/optic_plan_solver.cpp

```diff
@@ -0,0 +1,130 @@
+#include <sys/stat.h>
```

Comment:
There is a large amount of overlap between this file and popf_plan_solver.cpp. What do you think about creating abstract versions of the getPlan and check_domain functions in plansys2_core/src/plansys2_core/Utils.cpp? You could then just call those functions from the popf_plan_solver and optic_plan_solver plugins. That would probably make the code a bit more maintainable. For instance, in the future it might be a good idea to replace some of the line.find() commands with regular expressions in case some planners produce plans with slightly different syntax.

Reply:
Yes, I was going to start with this near-duplicate, then start a discussion on the PlanSys2 repo about creating a generic plugin that supports most planners that adhere to the pattern dictated by the IPC competition.   But since that could take a while to get approval and then make properly generic, this is used as a stop-gap.

Reply:
...but your idea is good because I've found several planner plugins out there, including another one for OPTIC that are already broken by small changes to the PlanSys2 core.   Having just one to maintain would result in fewer broken plugins.

Reply:
Cool. So the plan is to abstract this in future PRs, correct?

One other idea (for future improvement) ... Do we really need to write to temporary files and then make system calls to compute plans? Is there a more direct approach that could be applied, e.g., just pass the domain and problem strings to a function that would return the plan?

Reply:
I agree that it would be nice to find a different way of calling the planner than making system calls. A new issue seems like a good place to start that work.

Reply:
Yes, future PR, @jjzapf